### ISSUE TITLE
training hyper parameters

### ISSUE THREAD

**Issue by xiaovirgo** [JDAI-CV/DCL#19]
what's the meaning of the training hyper parameter '--cls_2' or '--cls_mul'? and what's the different between of them?

**Comment by akira-l** [comment]
classify layers are different as shown in LoadModel.py

**Comment by xiaovirgo** [comment]
yes,I also noticed this.But why set this difference? In zhe 'README.md','For training CUB / STCAR / AIR',he use the 'cls_mul','For training FGVC product datasets',he use the 'cls_2',I can't understand this.
They are all multi-classification problems.

**Comment by 361720691** [comment]
I also get some questions for the hyper parameters. In the 'README.md', the parameters in "For training FGVC product datasets from scratch" and "For training FGVC datasets from trained checkpoint" are the same, except the --start_epoch $LAST_EPOCH, is that means if start_epoch is 0, the code will train from scratch, if it's not 0, the code will try to train from the checkpoint I provided?

**Comment by akira-l** [comment]
cls_mul and cls_2 are not used for the main classifier. They are responsible for giving the distinguishable ability for networks which can tell organized and disorganized inputs. @xiaovirgo You may have a thorough understanding with complete code combined with the paper. 

The start_epoch, just as the literal meaning, sets the start point of training and also related to the decay step of the optimizer. @361720691 If start_epoch = 0, the optimizer is with the initial learning rate, which may not fix the lower learning rate adapted to learned checkpoints.

**Comment by tweetyone** [comment]
Same question with @xiaovirgo , why for training CUB / STCAR / AIR' use the 'cls_mul', for training FGVC product datasets' use the 'cls_2'? Is this a typo or a choice which have some reason?  
Cause I have seen [https://github.com/JDAI-CV/DCL/issues/2](url), may be you forget to change 'cls_mul' to 'cls_2'? @akira-l

**Comment by akira-l** [comment]
We encourage to use cls_2 in FGVC series dataset. FGVC product contains 2019 categories, and the butterfly has 5419. It is too costly to build up 2*2019 or 2*5419 fc, especially in such massive datasets.

**Comment by xiaovirgo** [comment]
> We encourage to use cls_2 in FGVC series datasets. FGVC product contains 2019 categories, and the butterfly has 5419. It is too costly to build up 2*2019 or 2*5419 fc, especially in such massive datasets.

got it!,thank you very much.I now use this code to train myselves dataset,it contains 1642 categories,I think i need use cls_2

**Comment by Dengshunge** [comment]
@akira-l if using cls_2 indicate whether the image swap or not, it use 2 channels. but cls_mul use 2*num_class, why?

**Comment by tranvnhan** [comment]
> @akira-l if using cls_2 indicate whether the image swap or not, it use 2 channels. but cls_mul use 2*num_class, why?

I also find it hard to understand `cls_mul`. We need a better explanation.
